gems/sorbet-runtime/test/types/props/decorator.rb

```diff
@@ -185,6 +185,7 @@ class InnerStruct < T::Struct
   end
 
   class OptionalArrayClass
+    include T::Props
```

Comment:
Why are these required now? Why wouldn't `T::Props::Serializable` still include `T::Props`?

Reply:
If we have `Serializable` include `T::Props`, then we can't include `Serializable` in the `T::ImmutableStruct` and thus lose the serialization features.

Reply:
but we could factor something out to like `T::Props::Private::SerializableImpl` and make

```ruby
module T::Props::Serializable
  include T::Props
  include T::Props::Private::SerializableImpl
end
```

right?

Reply:
Yes, we can do it. Be mindful that `Plugin` also included `T::Props`, which creates the same dependency for `WeakConstructor`, so we would need to do a similar split for those modules as well. Let me know if you agree with that direction.